Updated instructions on how to run tsan tests

dmitri-d opened the pull request:
Signed-off-by: Dmitri Dolguikh <ddolguik@redhat.com>

Updated developer dcos with details on how to execute tsan tests now that we have a tsan-instrumented libc++

jmarantz commented on bazel/README.md:
```diff
@@ -500,10 +500,11 @@ If you have clang-5.0 or newer, additional checks are provided with:
 
-Similarly, for [thread sanitizer (TSAN)](https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual) testing:
+[Thread sanitizer (TSAN)](https://github.com/google/sanitizers/wiki/ThreadSanitizerCppManual) tests rely on
+a TSAN-instrumented version of libc++ and have to be run under the docker sandbox:
```
There is no way to run tsan without docker now?

dmitri-d replied:
You should really use instrumented libc++ for that, or weird errors could be reported. It's possible to build an instrumented libc++ and use that locally, but that would require quite a bit more work.

jmarantz replied:
How much more work? Is it something other than downloading the libc++ sources and compiling it with some options?

I think that would be my preferred workflow as I've had a lot of trouble with docker.

mattklein123 replied:
Can we document both options?

dmitri-d replied:
It is downloading libc++ sources and building it, and then updating bazel config to point to it. 
One way to approach this would be to build libc++ from source as part of TSAN test run (all of this applies to MSAN or any other *SAN tests), but that would increase build time. Perhaps we could have both docker-sandbox and built-at-runtime config options?

jmarantz replied:
I would hope this would be part of the flow to set up Envoy development. Maybe optional.

lizan replied:
We can document / recommend some sort of "standard development environment", how to set up it on workstation / vm / docker, then how to run santizers in it. And not supporting sanitizers in other environment. WDYT?

mattklein123 replied:
I would probably just doc both options? I think some people might want to build their own libc++.

dmitri-d commented:
Aplogies for a forced pushed, I messed up local commit history.

* Added instructions on how to build tsan instrumented libc++ locally
* added a script to generate tsan config to execute tsan tests locally

jmarantz commented on bazel/README.md:
```diff
@@ -500,10 +500,44 @@ If you have clang-5.0 or newer, additional checks are provided with:
+
+cmake -GNinja -DLLVM_ENABLE_PROJECTS="libcxxabi;libcxx" -DLLVM_USE_LINKER=lld -DLLVM_USE_SANITIZER=Thread -DCMAKE_BUILD_TYPE=Release \
+  -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX="/opt/libcxx_tsan" "../llvm-project-llvmorg-10.0.0/llvm"
+ninja install-cxx install-cxxabi
```
fwiw this did not quite work for me; when I typed this I got:
```
ninja: error: loading 'build.ninja': No such file or directory
```

I couldn't find that underneath the tsan directory or ./llvm-project-llvmorg-10.0.0/llvm

jmarantz replied:
this doesn't need to block this PR though; clearly these instructions are on the right track and I"m fine iterating too.

dmitri-d replied:
@jmarantz: could you check what version of cmake you have installed? If it's not cmake3, could you install that and try again?

jmarantz replied:
Yeah...it's cmake 3.16.3, but I'll go ahead & merge for now. We can iterate on a solution for me after.
